include/drivers/flash.h

```diff
@@ -36,12 +36,20 @@ struct flash_pages_layout {
 };
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
+/**
+ * Flash memory parameters
+ */
+struct flash_parameters {
+	uint8_t erase_value; /* Byte value of erased flash */
+};
+
 typedef int (*flash_api_read)(struct device *dev, off_t offset, void *data,
 			      size_t len);
 typedef int (*flash_api_write)(struct device *dev, off_t offset,
 			       const void *data, size_t len);
 typedef int (*flash_api_erase)(struct device *dev, off_t offset, size_t size);
 typedef int (*flash_api_write_protection)(struct device *dev, bool enable);
+typedef const struct flash_parameters* (*flash_api_get_parameters)(void);
```

Comment:
Thinking again: this needs to take a `struct device *dev` parameter.  There's an outside chance of a generic flash driver for which different devices have different properties.

If we were not to do that (and I really do think we should) we could instead just store the pointer to the structure in the API structure, as is currently done for `write_block_size`.

Reply:
OK, I will change this.